ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -12847,13 +12849,21 @@ protected void saveViewDefinition() throws SemanticException {
     createVwDesc.setViewExpandedText(expandedText);
   }
 
-  private Set<String> getTablesUsed(ParseContext parseCtx) {
-    Set<String> tablesUsed = new HashSet<>();
+  private Set<SourceTable> getTablesUsed(ParseContext parseCtx) {
+    Set<SourceTable> tablesUsed = new HashSet<>();
     for (TableScanOperator topOp : parseCtx.getTopOps().values()) {
       Table table = topOp.getConf().getTableMetadata();
       if (!table.isMaterializedTable() && !table.isView()) {
         // Add to signature
-        tablesUsed.add(table.getFullyQualifiedName());
+        SourceTable sourceTable = new SourceTable();
```

Comment:
So I know the SourceTable file is a generated thrift file but it seems like we have soooo much code in SemanticAnalyzer and this code seems like it belongs in a different method somewhere.  Some kind of creator method that takes a Table as a parameter.   Maybe somewhere under standalone-metastore in a Factory class?  If we put it in there, we'd also have to pass in a boolean because standalone-metastore should not rely on AcidUtils

Reply:
Refactored how `SourceTable` instances are created and in some cases like the one you mentioned `TableName` is a better choice because the stats contained by `SourceTable` is not needed here.
Added factory methods to `ql.metadata.Table`
* `SourceTable createSourceTable()` - to create a `SourceTable` instance in an initial state referencing this table instance.
* `TableName getFullTableName()` - to create a `TableName` using this table instance.